### PR TITLE
feat(components, app): move ListItem and create a ListItem child

### PR DESCRIPTION
**Pull request by jerader**
closes AUTH-623 AUTH-628

# Overview

This PR moves the `ListItem` component in the app directory to the components directory and adds a media query to change the padding/border radius between ODD vs helix.

Additionally, a new list item child is created called `ListItemDescriptor` and 2 examples are added to the parent's story.

## Test Plan and Hands on Testing

Review code

Also, open the `ListItem` storybook and select the 2 other stories for the different types of `ListItemDescriptor`. You'll have to manually change the viewport size to reset it so you don't trigger the media query. Not sure how to fix that? 

## Changelog

- move `ListItem` and update affected component imports
- make new list item child
- add list item child to the list item story

## Review requests

see test plan

## Risk assessment

low

### CODE REVIEW

File: components/src/atoms/ListItem/ListItem.stories.tsx [hunk 1]
```diff
@@ -45,3 +45,29 @@ export const ListItem: Story = {
     ),
   },
 }
+
+export const ListItemDescriptorDefault: Story = {
```
Comment (jerader) [inline components/src/atoms/ListItem/ListItem.stories.tsx:49]: by default, the parameter viewport size is for the ODD, so you'll have to manually change the viewport size on the story when you view. I'm not sure if i can set the viewport size as a control type?

File: components/src/atoms/ListItem/index.tsx [hunk 2]
```diff
@@ -41,10 +45,16 @@ export function ListItem(props: ListItemProps): JSX.Element {
...
-      borderRadius={BORDERS.borderRadius12}
       onClick={onClick}
+      css={css`
+        padding: 0;
+        border-radius: ${BORDERS.borderRadius4};
+        @media ${RESPONSIVENESS.touchscreenMediaQuerySpecs} {
+          padding: ${SPACING.spacing16} ${SPACING.spacing24};
+          border-radius: ${BORDERS.borderRadius12};
+        }
+      `}
```
Comment (koji) [inline components/src/atoms/ListItem/index.tsx]: nit maybe create a new style, `LIST_ITEM_STYLE`?

```ts
const LIST_ITEM_STYLE = css`
  width: 100%;
  height: max-content;
  padding: 0;
  border-radius: ${BORDERS.borderRadius4};
  
  @media ${RESPONSIVENESS.touchscreenMediaQuerySpecs} {
     padding: ${SPACING.spacing16} ${SPACING.spacing24};
      border-radius: ${BORDERS.borderRadius12};
   }
`
```

File: components/src/atoms/ListItem/ListItemChildren/ListItemDescriptor.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  const { description, content, type } = props
+  return (
+    <Flex
+      flexDirection={DIRECTION_ROW}
+      gridGap={SPACING.spacing8}
+      width="100%"
+      justifyContent={type === 'mini' ? JUSTIFY_SPACE_BETWEEN : 'none'}
+      padding={
+        type === 'mini'
+          ? `${SPACING.spacing8} ${SPACING.spacing4}`
```
Comment (koji) [inline components/src/atoms/ListItem/ListItemChildren/ListItemDescriptor.tsx]: https://www.figma.com/design/WbkiUyU8VhtKz0JSuIFA45/Feature%3A-Protocol-Designer-Phase-1?node-id=4759-89854&m=dev

```suggestion
          ? `${SPACING.spacing4} ${SPACING.spacing8}`
```

File: components/src/atoms/ListItem/ListItemChildren/ListItemDescriptor.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+          : SPACING.spacing12
+      }
+    >
+      <Flex width={type === 'mini' ? 'auto' : '30%'}>{description}</Flex>
```
Comment (koji) [inline components/src/atoms/ListItem/ListItemChildren/ListItemDescriptor.tsx]:
```suggestion
      <Flex width={type === 'mini' ? FLEX_AUTO : '30%'}>{description}</Flex>
```

File: components/src/atoms/ListItem/index.tsx [hunk 5]
```diff
@@ -32,19 +33,31 @@ const LISTITEM_PROPS_BY_TYPE: Record<
...
+  ListItem is used in ODD and helix
+**/
 export function ListItem(props: ListItemProps): JSX.Element {
   const { type, children, onClick, ...styleProps } = props
   const listItemProps = LISTITEM_PROPS_BY_TYPE[type]
 
+  const LIST_ITEM_STYLE = css`
+    background-color: ${listItemProps.backgroundColor};
+    width: 100%;
+    height: max-content;
```
Comment (koji) [inline components/src/atoms/ListItem/index.tsx]:
```suggestion
    height: ${FLEX_MAX_CONTENT};
```

**Review by koji — APPROVED** [review]
left a couple of comments. 
once all of them are solved, this pr should be good to go.

**Pull request merged**